src/Dfc.CourseDirectory.Core/DataStore/Sql/QueryHandlers/GetLiveTLevelsReportHandler.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Collections.Generic;
+using System.Data.SqlClient;
+using Dapper;
+using Dfc.CourseDirectory.Core.DataStore.Sql.Queries;
+using Dfc.CourseDirectory.Core.Models;
+
+namespace Dfc.CourseDirectory.Core.DataStore.Sql.QueryHandlers
+{
+    public class GetLiveTLevelsReportHandler : ISqlAsyncEnumerableQueryHandler<GetLiveTLevelsReport, LiveTLevelsReportItem>
+    {
+        public async IAsyncEnumerable<LiveTLevelsReportItem> Execute(SqlTransaction transaction, GetLiveTLevelsReport query)
+        {
+            var sql = @$"
+SELECT      p.Ukprn,
+            p.ProviderName,
+            d.Name as TLevelName,
+            v.VenueName,
+            t.StartDate
+FROM        Pttcd.TLevels t with (nolock)
+INNER JOIN  Pttcd.TLevelLocations l with (nolock) ON l.TLevelId = t.TLevelId
```

Comment:
Is the `nolock` stuff necessary at this point?

Reply:
This is a potentially long running query that could lock the entire t-levels table if we omit the nolock hint for the duration.

Reply:
Is that the case for snapshot isolation?

Reply:
I thought we were now REPEATABLE READ?

Reply:
Reverted to snapshot now by default. From a quick test, snapshot won't block anything if used in a read capacity like above.

Reply:
What are the chances of that changing again in the future? I'll always lean towards caution on these things.

Reply:
We've used Snapshot forever apart from for a few weeks (where changing it caused more problems that it solved). Leaving `nolock` in seems fine.